components/sceneutil/attach.cpp

```diff
@@ -90,25 +86,25 @@ namespace SceneUtil
         std::string mFilter2;
     };
 
-    void mergeUserData(osg::UserDataContainer* source, osg::Object* target)
+    void mergeUserData(const osg::UserDataContainer* source, osg::Object* target)
     {
         if (!target->getUserDataContainer())
-            target->setUserDataContainer(source);
+            target->setUserDataContainer(osg::clone(source, osg::CopyOp::SHALLOW_COPY));
         else
         {
             for (unsigned int i=0; i<source->getNumUserObjects(); ++i)
```

Comment:
I think `source` can be null here as we're not checking it where we get it from. Previously it was just assigned to the target's user data container, so if it was null, the target would have no user data, but now we dereference it.

Reply:
Thanks. This is correct. There should have been this null check in my changes. Somehow it was not uploaded to github.

Reply:
Already take care of in master.